components/experimental/src/units/converter_ratio.rs

```diff
@@ -0,0 +1,56 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::units::ratio::IcuRatio;
+
+// TODO: add Mul & Add for references to avoid cloning.
+/// A trait for types that can be converted between two units.
+pub trait ConverterRatio: Clone {
+    fn reciprocal(&self) -> Self;
+
+    // TODO: remove this method once implement Mul & Add for references to avoid cloning.
+    fn add(&self, other: &Self) -> Self;
+
+    // TODO: remove this method once implement Mul & Add for references to avoid cloning.
+    fn mul(&self, other: &Self) -> Self;
+
+    // TODO: is there a way to implement this for all types that implement From<IcuRatio>?
+    fn from_icu_ratio(value: IcuRatio) -> Option<Self>;
```

Comment:
Can we keep `IcuRatio` an implementation detail as discussed last week? It can be `From<num_rational::BigRational>`

Reply:
We went back and forth on whether `IcuRatio` was public. I think you're right, we ended up with it being internal, in which case `From<num_rational::BigRational>` is more appropriate.

Reply:
Makes sense because it's imprecise. Don't use it as a super-trait then, but blanket implement your trait for `From<BigRational>`, and specialize `f64` (I think that is allowed?)?.

Reply:
```
impl Foo for T where T: From<BigRational> {}
impl Foo for f64 {}
```

Reply:
I will merge this PR and follow up with another PR to try this.

https://github.com/unicode-org/icu4x/issues/4847